### PR TITLE
Profile

### PR DESCRIPTION
**Pull request by tanhanliang**

### CODE REVIEW

File: app/User.php [hunk 1]
```diff
@@ -35,4 +35,8 @@ public function hasRole($role) {
         }
         return false;
     }
+
+    public function getModel(){
+        return Member::where('crsid',$this->crsid)->first();
```
Comment (wonglkd) [inline app/User.php]: You might want a space here! `('crsid', $this->crsid)`

Also, the idea here is correct! Laravel/Eloquent has something for this though, check out https://laravel.com/docs/5.3/eloquent-relationships#one-to-one

```
class User extends ...
{
    ....
    public function member()
    {
        return $this->hasOne('App\Member', 'crsid', 'crsid');
    }
}

```

File: app/Http/Controllers/Profile/ProfileController.php [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        
+        $user = Auth::user();
+        // get the first member where the crsid matches and update the model
+        $person = Member::where('crsid', $user->crsid)
+            ->first()
+            ->update($request->all());
+        return redirect()->route('member.profile')->with('alert-success', 'Profile updated.');
+    }
+    
+    public function profileRules(){
```
Comment (wonglkd) [inline app/Http/Controllers/Profile/ProfileController.php]: I still think that the rules should be merged (DRY philosophy - Don't Repeat Yourself), but I'm okay with it being deferred to a later re-factor in favour of moving fast and getting this working.

**Comment by wonglkd** [comment]
This looks much better than the previous one! I still have some suggestions for (future) improvements but those can come in a later commit - no point blocking this and delaying development - looks good to me, any opinions @spencerthang ?

**Comment by tanhanliang** [comment]
Not sure if taking the rules array from Member and unsetting keys is a good idea or not!

**Comment by wonglkd** [comment]
It's not great, but an improvement imo since it avoids copy-pasting code
(hurts maintainability, and introduces bugs whenever you forget to update
all the instances).

You could consider adding a flag to the method which you could check (like
how the strict parameter was added - that was something custom and not
inbuilt). That's a decent compromise, though not perfect.

**Comment by tanhanliang** [comment]
You mean like an integer flag with, say 0 for fresher rules and 1 for profile rules etc?

**Comment by wonglkd** [comment]
Could be a string too! (In other languages you would use enums but I think
a string is fine for PHP in the context of a small app)

**Comment by wonglkd** [comment]
Thanks for the quick changes! Looking better and better, just a couple more things that might manifest as bugs later if not resolved now...

**Comment by tanhanliang** [comment]
Yeah `member` works. The redirect worked when I tested it yesterday but on second thought i think its probably not a good idea to do that in a model.

**Comment by wonglkd** [comment]
looks good to me! merging it in. Thanks for your hard work! (:

**Pull request merged**